### PR TITLE
Add "CertificateManager" class along with tests

### LINKED ISSUE #16: Add `certificate manager` module in `auto-id` package
Need to add these functions:
- Initialize a certificate manager.
- prepare signing params
- Convert a subject name to a common name
- Pretty-print the certificate
- Certificate to PEM
- PEM to certificate
- Get subject common name
- Get certificate auto id
- Create CSR
- Sign CSR
- Create and Sign CSR
- Issue Certificate
- Self issue certificate
- Save certificate

### PR DESCRIPTION
**Pull request by abhi3700**
This PR adds `CertificateManager` class and its tests & resolves #16.

The main functions are:
- `prettyPrintCertificate`
- `createCSR`
- `signCSR`
- `createAndSignCSR`
- `issueCertificate`
- `selfIssueCertificate`
- `saveCertificate`

### CODE REVIEW

**Review by marc-aurele-besner — COMMENTED** [review]
Just one comment regarding the direct use of `fs`

File: packages/auto-id/src/certificateManager.ts [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    return certificate
+  }
+
+  saveCertificate(filePath: string): void {
+    if (!this.certificate) {
+      throw new Error('No certificate available to save.')
+    }
+
+    const certificatePem = CertificateManager.certificateToPem(this.certificate)
+    fs.writeFileSync(filePath, certificatePem, 'utf8')
```
Comment (marc-aurele-besner) [inline packages/auto-id/src/certificateManager.ts:326]: Why not use [save()](https://github.com/subspace/auto-sdk/blob/fa02528f2fdb887e30c4c96c1f20d8e34c81c546/packages/auto-utils/src/save.ts#L1) here?

**Review by marc-aurele-besner — APPROVED** [review]
LGTM 🚀

File: packages/auto-id/src/certificateManager.ts [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    const commonNameAttr = new AttributeTypeAndValue({
+      type: OID_COMMON_NAME,
+      value: subjectName,
+    })
+    return new x509.Name([[commonNameAttr]])
+  }
+
+  // protected static toCommonName(subjectName: string): JsonName {
+  //   return [{ '2.5.4.3': [subjectName] }] // OID for commonName
+  // }
```
Comment (jfrank-summit) [inline packages/auto-id/src/certificateManager.ts:72]: Is there a reason to leave this code?

File: packages/auto-id/src/certificateManager.ts [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+      )
+    }
+
+    // Prepare the certificate builder with information from the CSR
+    const notBefore = new Date()
+    const notAfter = new Date()
+    notAfter.setDate(notBefore.getDate() + validityPeriodDays)
+
+    let certificateBuilder = await x509.X509CertificateGenerator.create({
+      serialNumber: randomSerialNumber().toString(),
```
Comment (jfrank-summit) [inline packages/auto-id/src/certificateManager.ts:244]: You shouldn't need to generate a random serial number here. If omitted, the [x509 package](https://peculiarventures.github.io/x509/interfaces/X509CertificateCreateWithKeyParams.html#serialNumber) will do it for you.

**Review by jfrank-summit — COMMENTED** [review]
A couple of comments.

File: packages/auto-id/src/certificateManager.ts (hunk 2 above)
Comment (abhi3700) [reply packages/auto-id/src/certificateManager.ts:72]: Was using this as an alternative. Somehow missed removing it.

File: packages/auto-id/src/certificateManager.ts (hunk 3 above)
Comment (abhi3700) [reply packages/auto-id/src/certificateManager.ts:244]: I see.
Actually, i was using a custom function for this, while trying out the other libs for CertificateManager.
Removed now.

**Review by marc-aurele-besner — APPROVED** [review]

**Review by jfrank-summit — APPROVED** [review]

**Pull request merged**